### PR TITLE
[eslint] Allow `"latest"` as `ecmaVersion`

### LINKED ISSUE #13699: `@babel/eslint-parser` should support `ecmaVersion: latest`
### 💻

- [ ] Would you like to work on this feature?

### What problem are you trying to solve?

From [`eslint@7.30`](https://eslint.org/blog/2021/07/eslint-v7.30.0-released), `eslint` default parser supports `latest` as `ecmaVersion`, but `@babel/eslint-parser` does not support it. Seems it should be fixed.

### Describe the solution you'd like

^

### Describe alternatives you've considered

^

### Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by fisker**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Closes #13699
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      | Yes
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

Ref eslint/eslint#14720

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/13638"></a>

### CODE REVIEW

File: eslint/babel-eslint-parser/src/configuration.cjs [hunk 1]
```diff
@@ -11,7 +11,7 @@ exports.normalizeESLintConfig = function (options) {
 
   return {
     babelOptions: { cwd: process.cwd(), ...babelOptions },
-    ecmaVersion,
+    ecmaVersion: ecmaVersion === 'latest' ? 2020 : ecmaVersion,
```
Comment (nicolo-ribaudo) [inline eslint/babel-eslint-parser/src/configuration.cjs:14]: Does it throw if we just directly pass `"latest"`?

Comment (fisker) [reply eslint/babel-eslint-parser/src/configuration.cjs:14]: I don't know, I can't find where it's used.

Comment (fisker) [reply eslint/babel-eslint-parser/src/configuration.cjs]: But yes, if we use `"latest"`, the parser throws https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1465/checks?check_run_id=3248815028#step:5:18825

Comment (fedeci) [reply eslint/babel-eslint-parser/src/configuration.cjs]: `eslint-scope` does not support `'latest'` as valid. Does is make more sense to update it there?

File: eslint/babel-eslint-parser/src/configuration.cjs [hunk 2]
```diff
@@ -11,7 +11,7 @@ exports.normalizeESLintConfig = function (options) {
 
   return {
     babelOptions: { cwd: process.cwd(), ...babelOptions },
-    ecmaVersion,
+    ecmaVersion: ecmaVersion === "latest" ? 2020 : ecmaVersion,
```
Comment (JLHwung) [inline eslint/babel-eslint-parser/src/configuration.cjs:14]: The latest is 2022 as of August 2021. I think we can normalize `"latest"` to `1e8` like acorn did (https://github.com/acornjs/acorn/blob/691a31b1e3d6936f4e6991da3d81c1a6b06610cf/acorn/src/options.js#L109) so we don't have to update it every year.

File: eslint/babel-eslint-parser/src/configuration.cjs (hunk 1 above)
Comment (fisker) [reply eslint/babel-eslint-parser/src/configuration.cjs]: `eslint-scope` suggest handling on parser side. https://github.com/eslint/eslint-scope/issues/74#issuecomment-898780125

File: eslint/babel-eslint-parser/src/configuration.cjs (hunk 2 above)
Comment (fisker) [reply eslint/babel-eslint-parser/src/configuration.cjs:14]: Good idea.

**Review by fedeci — APPROVED** [review]

**Review by nicolo-ribaudo — APPROVED** [review]

File: eslint/babel-eslint-parser/src/configuration.cjs (hunk 2 above)
Comment (fisker) [reply eslint/babel-eslint-parser/src/configuration.cjs]: I won't miss the date.

**Pull request merged**